src/services/permissions-checker.js

```diff
@@ -29,7 +32,81 @@ function PermissionsChecker(
     return allowed && (!users || users.includes(parseInt(userId, 10)));
   }
 
-  function isAllowed() {
+  // NOTICE: Compute a scope to replace $currentUser variables with
+  //         the actual user values. This will generate the expected
+  //         conditions filters when applied on the server scope response
+  function computeConditionFiltersFromScope(userId, collectionListScope) {
+    const computedConditionFilters = _.clone(collectionListScope.filter);
+    computedConditionFilters.conditions.forEach((condition) => {
+      if (condition.value
+        && condition.value.startsWith('$')
+        && collectionListScope.dynamicScopesValues.users[userId]) {
+        condition.value = collectionListScope
+          .dynamicScopesValues
+          .users[userId][condition.value];
+      }
+    });
+    return computedConditionFilters;
+  }
+
+  // NOTICE: Check if `expectedConditionFilters` at least contains a definition of
+  //         `actualConditionFilters`
+  function isConditionFromScope(expectedFilterConditions, actualFilterCondition) {
+    return expectedFilterConditions.filter((expectedCondition) =>
+      expectedCondition.value === actualFilterCondition.value
+      && expectedCondition.operator === actualFilterCondition.operator
+      && expectedCondition.field === actualFilterCondition.field).length > 0;
+  }
+
+  async function isCollectionListAllowed(collectionListScope) {
+    if (!collectionListScope) {
+      return true;
+    }
+
+    try {
+      const expectedConditionFilters = computeConditionFiltersFromScope(
+        collectionListRequest.userId,
+        collectionListScope,
+      );
+
+      const formatAggregation = (aggregator, conditions) => {
```

Comment:
I think you can add a notice here to explain what the function do or rename the function, the name is not relevant IMO.